plugins/in_splunk/splunk_prot.c

```diff
@@ -0,0 +1,822 @@
+{
+    int i;
+    int ret;
+    int len;
+    char *uri;
+    char *qs;
+    off_t diff;
+    flb_sds_t tag;
+    struct mk_http_header *header;
+
+    if (request->uri.data[0] != '/') {
+        send_response(conn, 400, "error: invalid request\n");
+        return -1;
+    }
+
+    /* Decode URI */
+    uri = mk_utils_url_decode(request->uri);
+    if (!uri) {
+        uri = mk_mem_alloc_z(request->uri.len + 1);
+        if (!uri) {
+            return -1;
+        }
+        memcpy(uri, request->uri.data, request->uri.len);
+        uri[request->uri.len] = '\0';
+    }
+
+    /* Try to match a query string so we can remove it */
+    qs = strchr(uri, '?');
+    if (qs) {
+        /* remove the query string part */
+        diff = qs - uri;
+        uri[diff] = '\0';
+    }
+
+    /* Refer the tag at first*/
+    if (ctx->ins->tag && !ctx->ins->tag_default) {
+        tag = flb_sds_create_size(1024);
+        if (!tag) {
+            return -1;
+        }
+        /* clear tag before using */
+        flb_sds_len_set(tag, 0);
+        flb_sds_cat(tag, ctx->ins->tag, flb_sds_len(ctx->ins->tag));
+    }
+    else {
+        /* Compose the query string using the URI */
+        len = strlen(uri);
+
+        if (len == 1) {
+            tag = NULL; /* use default tag */
+        }
+        else {
+            tag = flb_sds_create_size(len);
+            if (!tag) {
+                mk_mem_free(uri);
+                return -1;
+            }
+
+            /* New tag skipping the URI '/' */
+            flb_sds_cat(tag, uri + 1, len - 1);
+
+            /* Sanitize, only allow alphanum chars */
+            for (i = 0; i < flb_sds_len(tag); i++) {
+                if (!isalnum(tag[i]) && tag[i] != '_' && tag[i] != '.') {
+                    tag[i] = '_';
+                }
+            }
+        }
+    }
+
+    /* Check if we have a Host header: Hostname ; port */
+    mk_http_point_header(&request->host, &session->parser, MK_HEADER_HOST);
+
+    /* Header: Connection */
+    mk_http_point_header(&request->connection, &session->parser,
+                         MK_HEADER_CONNECTION);
+
+    /* HTTP/1.1 needs Host header */
+    if (!request->host.data && request->protocol == MK_HTTP_PROTOCOL_11) {
+        flb_sds_destroy(tag);
+        mk_mem_free(uri);
+
+        return -1;
+    }
+
+    /* Should we close the session after this request ? */
+    mk_http_keepalive_check(session, request, ctx->server);
+
+    /* Content Length */
+    header = &session->parser.headers[MK_HEADER_CONTENT_LENGTH];
+    if (header->type == MK_HEADER_CONTENT_LENGTH) {
+        request->_content_length.data = header->val.data;
+        request->_content_length.len  = header->val.len;
+    }
+    else {
+        request->_content_length.data = NULL;
+    }
+
+    if (request->method == MK_METHOD_GET) {
+        /* Handle health minotoring of splunk hec endpoint for load balancers */
+        if (strncmp(uri, "/services/collector/health", 26) == 0) {
```

Comment:
`strncasecmp` or maybe just `strcasecmp` because `uri` is NULL terminated and hardcoding the length makes no sense in these cases

Reply:
Sure. I've fixed it in https://github.com/fluent/fluent-bit/pull/7349/commits/e1eb2a517eae3d17099d8e1336770225bb42d290/